### ISSUE TITLE
Website Concern for Ticket

### ISSUE THREAD

**Issue by C13L0** [phsc-edu/2018#123]
This is a two part ticket. 
-----

@curtiskphsc assigning this part to you. I recommend adding this to an FAQ for testing services.
---------------
For example, students do not know to select, “Quick Links” then “Testing Services” to find information on the PERT.

@jeppy64 This will be part of phase two.
It has been requested that the main site to have multi site search enabled. We have apache solr available to our main site which would allow this functionality.
--------------
The search bar continues to not function. 
The search should be able to pull information from all of the subdomains so that students may locate answers to their questions.

**Comment by C13L0** [comment]
The prototype search has been completed at https://dev-phsc-advising.pantheonsite.io/

In order to test, search as you normally would by clicking on the magnifying icon and typing in a search word.

It will then bring you to a page which by default searches that website. So in this instance, the word that was typed in will search Advising only.

On the same page is a new google search which searches all PHSC websites. The search term will have to be re-typed as there is not a way to inject the search term into the google search.

Assigning to @10yson for testing

**Comment by C13L0** [comment]
@10yson @curtiskphsc 

> Search results are displaying Google Ads. Can the ads be eliminated (or hidden) from search results?

Yes!  I will work with Rick to submit the request to to Google.
```
Google For Education, https://edu.google.com/products/productivity-tools/, to offer Custom Search Engine on your website for free without advertising.
```

**Comment by C13L0** [comment]
@10yson @curtiskphsc 
> After I use the global search, the search results are displayed as expected. However, I did not see a place for me to modify/refine my query for another global search.

This looks like a javascript error and will take time to troubleshoot as it could be an issue with the module, core javascript, or even the theme. If it can't be solved, we can add a link to direct the user back to the main search page.

**Comment by C13L0** [comment]
@10yson @curtiskphsc 

> On my phone, when switching to the global search, the Drupal search box in the page header is still very prominent on the page. Visually, it wasn't immediately intuitive where to enter my search keywords although I easily figured it out.

We could remove the search in the header on the search page entirely. This would be a theme layout and design change.

**Comment by C13L0** [comment]
Assigning this ticket to @curtiskphsc for a search page layout and design.

**Comment by curtiskphsc** [comment]
@C13L0 Just a thought. Has this been tested using Microsoft Edge? Microsoft is shipping newer machines with Edge as the default browser, not IE 11.

**Comment by C13L0** [comment]
@curtiskphsc I don't have a way to test on Edge but the website itself has been tested on Edge, so it should be fine. If you install a chrome adblocker you can view it without ads. I would recommend adblock plus.

**Comment by C13L0** [comment]
@10yson @curtiskphsc 

I removed the search bar from the header on all search pages. This should help reduce confusion on mobile and desktop.

**Comment by C13L0** [comment]
@curtiskphsc 

Dr. Harts has advised me that the design needs to be implemented asap per her conversation with Lucy Miller and Dr. Beard . She informed me that he knows it's ready and she will be sending it to him for review. Please have the layout by friday.

**Comment by C13L0** [comment]
@curtiskphsc As per Dr. Harts, this is a rush item. Please see screenshots of sections that need design/layout.

**Search Links**
![search-links](https://user-images.githubusercontent.com/3138019/39484471-e9e42802-4d43-11e8-9855-bbd73f3b1d65.png)

**Search page**
![search-input-page](https://user-images.githubusercontent.com/3138019/39484494-fc87b58c-4d43-11e8-81de-13aaaf59baaa.png)

**Search results**
![search-results](https://user-images.githubusercontent.com/3138019/39484523-11779732-4d44-11e8-8d87-c8c8c777562e.png)

**Search again** (only on google search page)
![search-again](https://user-images.githubusercontent.com/3138019/39484574-2b6831b0-4d44-11e8-8a05-ed5b3e4dc48e.png)

**Comment by C13L0** [comment]
@curtiskphsc Excellent! I look forward to it :D

**Comment by curtiskphsc** [comment]
@C13L0 Could you please provide a link to the Drupal module being utilized for this?

**Comment by C13L0** [comment]
@curtiskphsc 

At this time, we are not able to use Apache Solr for multisite search due to a dependency on Composer. We would have to rebuild each website from the ground up, as confirmed by Pantheon.

> On Tue, Apr 24, 2018 at 10:13 AM, "Cielo Johnson" wrote: 
> Good morning, I am looking for clarification of adding Apache Solr to a current drupal 8 site.
> All of the directions that I have found walk you through building a *new* composer managed site
> **Does this mean that I need to rebuild the website from scratch in order to add composer?**
> https://pantheon.io/docs/solr-drupal-8/

> On Tue, Apr 24, 2018 at 10:21 AM, "Ruthie Hallarsis" wrote: 
> Hello Cielo! Ruthie here. Yes that is correct. That is because "Since one module relies on Solarium, an external library, in addition to Composer's autoloader, we cannot support non-Composer workflows for Solr on Drupal 8" as stated there in the doc you posted. For more information about that you can also check this issue from drupal.org: https://www.drupal.org/node/2858750

The good news is that a patch is currently being worked on which will allow us to add Solr without composer. Once the patch is committed, we can continue with our efforts of adding multi-site search via Solr to our sites.  https://www.drupal.org/project/search_api_solr/issues/2891694#comment-12597753

As a temporary solution, we are using a custom google search on every website. Jeff House added our sites to our Google for Education account and all ads have been removed. 

Dr. Harts has confirmed with Dr. Beard that we will be going live with the custom google search on all sites next week.

**Comment by longc77** [comment]
@C13L0 @curtiskphsc Thank you for the impromptu meeting yesterday Cielo!

To summarize, here’s an outline of what we discussed:
1. CL to provide an edited wireframe of a subdomain page to include theming for:
    a.	Search “Subdomain name” and Search all PHSC sites in the menu bar
    b.	Search bar
    c.	Search results
    d.	Also mentioned, that Cielo’s not sure if the “go” button once clicking Search all PHSC sites from a subdomain can be modified. It’s part of that Google Search function. CL will include it in the mock to be safe.
    e.	Discussed making the Search Again link prominent in theme

2. Marketing to evaluate the magnifying icon, in menu bar, on a phsc.edu and determine if it is now appropriate with the changes being made to the subpage domains.
    a.	CL to provide revised wireframe if needed
    b.	Search results not needed–Cielo will do her best to match the theme from the subdomain wireframe but she’s not sure how much theming can be done to the Google module results page.
    c.	Cielo will make Search Again link theme consistent 

3. We understand that Drupal will not provide support for any security issues with the Simple Google Custom Search Engine Module in its current form. https://www.drupal.org/project/simple_gse_search – Please confirm link is correct and continue to keep us posted on the development.

4. We also understand that this is temporary in nature until the Drupal 8 Core Search Module can be used as originally intended.

5. Going forward, we would like to be invited to meetings involving the website. For me, being included in such meetings can help obtain a better understanding of the process or spark some type of creative “A-ha” moment by just being in the room. 

6. After speaking to Lucy, we will produce the above materials. However, with our team’s involvement today in commencement, the photo booth and commercial filming, we have lost a day of production. The materials that are completed by EOD Friday will be uploaded to the ticket for Cielo. If an item is not completed, we will begin working on it first thing Monday, May 7. We are committed to working together to make the site better.

**Comment by C13L0** [comment]
@longc77
> We are committed to working together to make the site better.

Corey, thank you for your commitment. Your attitude and positiveness is awesome. :D

**Comment by longc77** [comment]
Hello @C13L0 @curtiskphsc! I’ve put together the wireframes as discussed. Included in the PDF are two options.
 
The Marketing team recommends Option B. Option B is simple, provides a seamless transition and would be user friendly. An external visitor to the site, wants to find the information they are looking for fast. This option allows them to do so.
 
Option A, has the potential to cause confusion for the end user. Users may not understand the difference between the 2 search options and accidentally click one and not the other. It’s not necessary for the user to know that there are 22 different subdomains.

From a marketing perspective of offering the user the best possible experience on our site, we would like to see the search function search all pages from the beginning, giving the user all of the results containing they key word or phrase in a priority listing such as the example below.
 
(Example: http://search.usf.edu/search?q=admission&btnG=Google+Search&btng=Go&client=tampa&proxystylesheet=usf-edu-cms-v2&output=xml_no_dtd&numgm=5&sort=date%3AD%3AL%3Ad1&wc=200&wc_mc=1&oe=UTF-8&ie=UTF-8&ud=1&exclude_apps=1&site=tampa)

Please let me know if you need anything else on this. Thanks, Corey

**Comment by longc77** [comment]
@C13L0, can you see the PDF?

**Comment by longc77** [comment]
@C13L0, can you see the PDF?

**Comment by longc77** [comment]
@C13L0, can you see the PDF?

**Comment by C13L0** [comment]
@longc77  Looks great! Due to current restrictions using a custom google search, we are going to temporarily use Option A for theming. Once we can create our own custom search with apache solr, then I will re-theme the search to Option B. I appreciate your time coming up with both.

**Comment by C13L0** [comment]
@longc77 

It looks like there are font size changes for Pasco-Hernando State College and subdomain name. If this is correct, can you please provide the font sizes?

> Fill: is there a default or do I need to provide this color to you?
The fill can be any color you like as long as it's ada compliant! Please provide the color you would like.

![font-size](https://user-images.githubusercontent.com/3138019/39646722-5ec5c550-4faa-11e8-84e9-7a05a7012877.png)

**Comment by longc77** [comment]
Hi @C13L0. I grabbed it from one of the last wireframes I had on file. Weird. Not sure how we're off. Since we never had a chance to sit down together and tweak for style, please make it consistent with what you have now. TY!

For the fill, can you make it a percentage of the dark grey that's park of the style guide (#A7A9AC color used for breadcrumbs)?

**Comment by C13L0** [comment]
Yes, I can make it a specific percentage. Do you have one in mind or should I eyeball it?

**Comment by C13L0** [comment]
@longc77 Yes, I can make it a specific percentage. Do you have one in mind or should I eyeball it?

**Comment by longc77** [comment]
Good morning @C13L0. Did you eyeball it or can I still provide a specific %? If you did, can I take a look at what you have?

**Comment by C13L0** [comment]
@longc77 I haven't pushed the changes up yet, so you can still make adjustments =)

**Comment by longc77** [comment]
K. I'm thinking 20% of the #58595B (on style guide) would look good. If there are any concerns with the color, please let me know. Thanks @C13L0

**Comment by longc77** [comment]
@C13L0, from your comment before (copied and pasted below), 
I don't understand what that means. Please explain in layman terms – explain current restrictions, etc. Is our own custom search timeline TBD? TY! 

C13L0 commented May 3, 2018 • edited 
@longc77 Looks great! Due to current restrictions using a custom google search, we are going to temporarily use Option A for theming. Once we can create our own custom multi-site search with apache solr, then I will re-theme the search to Option B. I appreciate your time coming up with both.

**Comment by longc77** [comment]
@C13L0, based on our conversation this AM, how are these?

**Comment by C13L0** [comment]
@longc77 awesome! thanks =)

**Comment by longc77** [comment]
Notes from mock wireframes on programming that can't be done at this time:
-search bar where you enter terms can not exist (only the drop down menu)
-it has to take you to another page to enter your search terms
-search results word/phrase can't be highlighted on Google page
-search results word/phrase can be highlighted on Drupal page and is set up that way but is not functioning that way. Jay and Cielo are looking into it.
-search results Drupal page can't have a drop down to sort by relevance or date
-search results Google page can have a drop down to sort by relevance or date
-mentioned we would like the main search drop down menu, in the future, to include the directory and possibly other pages/choices specifically

**Comment by jeppy64** [comment]
Time to complete: 12 hours - all sites

All sub domains have:
Local site search
Global search through Google Custom search
core Relativity content types, plus individual specialty content types

Marketing Site has:
Global search through Google Custom search

**Comment by C13L0** [comment]
@jeppy64 Signing off on this as completed. 

I created a new ticket to complete the subdomain search menu when @longc77 and I have a chance to work on it.

**Comment by curtiskphsc** [comment]
@C13L0 The universal search is only returning one page from the Catalog no matter the search term entered, and that page has nothing to do with the term searched.  See screenshot. Searched Unmanned Vehicle Systems Operation and the only catalog page returned is the Dental Hygiene page?
![search-function-issue](https://user-images.githubusercontent.com/28155245/40139951-83b26d0e-591f-11e8-8b1d-95ee89be396f.JPG)

**Comment by curtiskphsc** [comment]
@C13L0 Searched Cybersecurity and returned Paramedic page. 
![cybersecurity-search](https://user-images.githubusercontent.com/28155245/40140352-b57c78ec-5920-11e8-9574-55fdf44d19f6.JPG)

**Comment by longc77** [comment]
@C13L0 As mentioned in **Theme subdomain search menu at mobile #213:**
"@longc77 Due to time constraints, I themed the search menu as option one." 
* The site doesn't match the mock provided. The text in the search is black and non bold. The one on the mock is white and bold. 
* The search results links are not the correct color
* where's the search again button - when we search from phasic.edu and don't type in a word in the search bar a dead page comes up that says "Search results" where the R in results needs to be capped. 
*When we search a term that's not a part of our library, it goes to a weird intermediate java script message page for a split second. 

I do see that the words are highlighted in both search options and you can search and you can use the search bar to search again (on subdomains). **THANK YOU @C13L0  !**

Is the drop down menu still possible and if so, how long would that take? Is it part of the plan to implement at any point going forward?

*Note: I know time restrictions were mentioned but Marketing is responsible for how the site is to look. This is not look we are going for. After discussing options with Cielo, I went back and spent more time on wireframes (about 3 hours) toying around on how I wanted the drop down search to look. For being truly customizable, I feel like design wise, it's not. Feeling pigeon holed as a designer due to functionality. Although not our preferred design, we did our best within parameters provided by the Webmaster to make it look good with what was already in place.

**Comment by C13L0** [comment]
@curtiskphsc I'm not able to customize the google search any further.
